SimFastTiming/FastTimingCommon/src/ETLDeviceSim.cc

```diff
@@ -21,17 +34,38 @@ void ETLDeviceSim::getHitsResponse(const std::vector<std::tuple<int,uint32_t,flo
     const uint32_t id  = std::get<1>(hitRefs[i]);
     const MTDDetId detId(id);
 
-    // Safety check
-    if ( detId.det()!=DetId::Forward || detId.mtdSubDetector()!=2 ) continue;
-
-    auto simHitIt = simHitAccumulator->emplace(id,mtd_digitizer::MTDCellInfo()).first;
-      
+    // Safety check (this should never happen, it should be an exception
+    if ( detId.det()!=DetId::Forward || detId.mtdSubDetector()!=2 ) {
+      throw cms::Exception("ETLDeviceSim") << "got a DetId that was not ETL!!!";
```

Comment:
print the DetId in the exception message

Reply:
Done